src/main/clojure/clara/rules/durability/fressian.clj

```diff
@@ -132,7 +132,11 @@
   ([^Reader rdr]
    (read-record rdr nil))
   ([^Reader rdr add-fn]
-   (let [builder (-> (.readObject rdr) resolve deref)
+   (let [try-resolve #(if-let [clazz (resolve %)]
+                        clazz
```

Comment:
I don't think this is a class, but rather a Clojure Var for the record constructor function.  From the implementation before this PR:

````
(defn write-record
  "Same as write-with-meta, but with Clojure record support.  The type of the record will
   be preserved."
  [^Writer w tag rec]
  (let [m (meta rec)]
    (.writeTag w tag 3)
    (.writeObject w (record-map-constructor-name rec) true)
    (write-map w rec)
    (if m
      (.writeObject w m)
      (.writeNull w))))
````

Then looking at the current read implementation:

```
(defn read-record
  "Same as read-with-meta, but with Clojure record support.  The type of the record will
   be preserved."
  ([^Reader rdr]
   (read-record rdr nil))
  ([^Reader rdr add-fn]
   (let [builder (-> (.readObject rdr) resolve deref)
         build-map (.readObject rdr)
         m (read-meta rdr)]
     (cond-> (builder build-map)
       m (with-meta m)
       add-fn add-fn))))
````
If the resolved value was a class it wouldn't be possible to invoke it as a Clojure function - note the ```(builder build-map)``` call.

Assuming I'm not missing something, this makes testing easier if you're so inclined, though for this sort of error handling I don't think it is critical.  You could just use [ns-unmap](https://clojuredocs.org/clojure.core/ns-unmap) before deserializing.

Reply:
This is a good point. So the suggestion (besides testing) is that the resolved symbol isn't a class at all and the message should avoid being worded suggesting that it is. Also, we shouldn't call it `clazz`, but perhaps just `resolved` or `resolved-var` or something like that? It is notable that `deref` is called on it a step later, so it definition can't be a class.

Reply:
corrected:
https://github.com/cerner/clara-rules/pull/476/commits/5252ca54994b2a521897bce2a6d8a2af5b223edf

Reply:
todo: 
add a test